closure/goog/testing/asserts_test.js

```diff
@@ -436,9 +438,61 @@ function testAssertObjectEqualsMap() {
   assertObjectEquals('maps should be equal', map1, map2);
 
   map1.set('hi', 'hey');
-  assertThrows('sets should not be equal', function() {
-    assertObjectEquals(map1, map2);
-  });
+  assertThrows('maps should not be equal',
+               function() { assertObjectEquals(map1, map2); });
+}
+
+
+function testAssertObjectEqualsTypedArray() {
+  var typedArrays = [
+    'Float32Array',
+    'Float64Array',
+    'Int8Array',
+    'Int16Array',
+    'Int32Array',
+    'Uint8Array',
+    'Uint8ClampedArray',
+    'Uint16Array',
+    'Uint32Array',
+    'BigInt64Array',
+    'BigUint64Array',
+  ];
+
+  var prior, priorName;
+  for (var i = 0; i < typedArrays.length; ++i) {
+    var name = typedArrays[i];
+    var arrType = window[name];
+    if (arrType) {
+      function construct(num) { return num; }
+      function mutate(array, offset) { array[0] *= -1; }
+      if (name.startsWith('Big')) {
+        construct = function(num) { return BigInt(Math.floor(num)); };
+        mutate = function(array) { array[0] += BigInt(Math.pow(2, 32)); };
+      }
+
+      // Allocate shifted buffers.
+      var buf1 = new ArrayBuffer(3 * arrType.BYTES_PER_ELEMENT);
+      var buf2 = new ArrayBuffer(4 * arrType.BYTES_PER_ELEMENT);
+      var arr1 = new arrType(buf1, 0, 3);
+      var arr2 = new arrType(buf2, arrType.BYTES_PER_ELEMENT, 3);
+
+      // Fill them with the same contents.
+      arr1[0] = arr2[0] = construct(1);
+      arr1[1] = arr2[1] = construct(2.5);
+      arr1[2] = arr2[2] = construct(-3);
+      assertObjectEquals(name + 's should be equal', arr1, arr2);
+
+      mutate(arr2);
+      assertThrows(name + 's should not be equal',
```

Comment:
I think this needs to be assertThrowsJsUnitException.  I see that we don't have that elsewhere in this file, which leads me to believe we turn off check that it enables globally, but it still makes sense to do the right thing here.